src/Avalonia.Base/Styling/ThemeVariant.cs

```diff
@@ -100,11 +100,11 @@ public bool Equals(ThemeVariant? other)
         {
             PlatformThemeVariant.Light => Light,
             PlatformThemeVariant.Dark => Dark,
-            _ => throw new ArgumentOutOfRangeException(nameof(themeVariant), themeVariant, null)
+            _ => Default,
```

Comment:
nit: As far as I can see, this branch will never be executed, non-nullable PlatformThemeVariant should have only two values.